samples/bluetooth/peripheral_ht/src/hts.c

```diff
@@ -53,26 +56,64 @@ BT_GATT_SERVICE_DEFINE(hts_svc,
 
 void hts_init(void)
 {
+	temp_dev = device_get_binding("TEMP_0");
+
+	if (!temp_dev) {
+		printk("error: no temp device\n");
+		return;
+	}
+
+	printk("temp device is %p, name is %s\n", temp_dev,
+	       temp_dev->config->name);
 }
 
 void hts_indicate(void)
 {
 	/* Temperature measurements simulation */
+	struct sensor_value temp_value;
+
 	if (simulate_htm) {
+		static u8_t htm[5];
+		u32_t mantissa;
+		u8_t exponent;
+		u32_t *ieee;
+		int r;
+
 		if (indicating) {
 			return;
 		}
 
-		temperature++;
-		if (temperature == 30U) {
-			temperature = 20U;
+		if (!temp_dev) {
+			temperature++;
+			if (temperature == 30U) {
+				temperature = 20U;
+			}
+
+			goto gatt_indicate;
+		}
+
+		r = sensor_sample_fetch(temp_dev);
+		if (r) {
+			printk("sensor_sample_fetch failed return: %d\n", r);
+		}
+
+		r = sensor_channel_get(temp_dev, SENSOR_CHAN_DIE_TEMP,
+				       &temp_value);
+		if (r) {
+			printk("sensor_channel_get failed return: %d\n", r);
 		}
 
-		static u8_t htm[1 + 4];
-		htm[0] = 0x0; /* uint8, temperature in celsius*/
-		/* (IEEE-11073 32-bit FLOAT). */
-		htm[1] = temperature; /* mantissa = temperature*/
-		htm[2] = htm[3] = htm[4] = 0; /* sign = 0, exponent= 0 */
+		temperature = sensor_value_to_double(&temp_value);
+
+gatt_indicate:
+		printf("temperature is %gC\n", temperature);
+
+		mantissa = (u32_t)(temperature * 100);
+		exponent = (u8_t)-2;
+
+		htm[0] = 0; /* temperature in celcius */
+		ieee = (u32_t *)&htm[1];
+		*ieee = ((u32_t)exponent << 24) | mantissa;
```

Comment:
How is this alignment-safe? Seems like you should be using `UNALIGNED_PUT()`

Reply:
can i use sys_put function for unaligned memory location ?

Reply:
Asking the right question! :)

Yes, you should use sys_put_le and sys_get_le, because the HTS specification states that little endian is the byte transmission order. There is also sys_put_le24 which would come in handy in this case.

You can check that it works with the nRF Connect for Mobile App.

Reply:
As u said , i used sys function and it works with nRF connect as well as with central_ht